qutebrowser/mainwindow/tabwidget.py

```diff
@@ -736,9 +730,17 @@ def _get_icon_rect(self, opt, text_rect):
                      else QIcon.Disabled)
         icon_state = (QIcon.On if opt.state & QStyle.State_Selected
                       else QIcon.Off)
-        tab_icon_size = opt.icon.actualSize(icon_size, icon_mode, icon_state)
-        tab_icon_size = QSize(min(tab_icon_size.width(), icon_size.width()),
-                              min(tab_icon_size.height(), icon_size.height()))
+        # reserve space for favicon when tab bar is vertical (issue #1968)
+        position = config.get('tabs', 'position')
+        show_favicons = config.get('tabs', 'show-favicons')
+        if (opt.icon.isNull()
+            and position in [QTabWidget.East, QTabWidget.West]
+            and show_favicons):
+            tab_icon_size = icon_size
+        else:
+            actual_size = opt.icon.actualSize(icon_size, icon_mode, icon_state)
```

Comment:
I saved actual size of favicon to actual_size rather than tab_icon_size just to shorten the line (otherwise it was too long).